openmetadata-service/src/main/java/org/openmetadata/service/events/scheduled/EventSubscriptionScheduler.java

```diff
@@ -156,62 +164,196 @@ public void deleteEventSubscriptionPublisher(EventSubscription deletedEntity)
-    if (eventSubscription == null) {
-      EntityRepository<? extends EntityInterface> subscriptionRepository =
-          Entity.getEntityRepository(Entity.EVENT_SUBSCRIPTION);
-      EventSubscription subscription =
-          (EventSubscription)
-              subscriptionRepository.get(
-                  null, subscriptionId, subscriptionRepository.getFields("id"));
-      if (subscription != null && (Boolean.FALSE.equals(subscription.getEnabled()))) {
-        return new SubscriptionStatus().withStatus(SubscriptionStatus.Status.DISABLED);
-      }
+    Optional<EventSubscription> eventSubscriptionOpt =
+        getEventSubscriptionFromScheduledJob(subscriptionId);
+
+    if (eventSubscriptionOpt.isPresent()) {
+      return eventSubscriptionOpt.get().getDestinations().stream()
+          .filter(destination -> destination.getId().equals(destinationId))
+          .map(SubscriptionDestination::getStatusDetails)
+          .findFirst()
+          .orElse(null);
+    }
+
+    EntityRepository<? extends EntityInterface> subscriptionRepository =
+        Entity.getEntityRepository(Entity.EVENT_SUBSCRIPTION);
+
+    // If the event subscription was not found in the scheduled job, check the repository
+    Optional<EventSubscription> subscriptionOpt =
+        Optional.ofNullable(
+            (EventSubscription)
+                subscriptionRepository.get(
+                    null, subscriptionId, subscriptionRepository.getFields("id")));
+
+    return subscriptionOpt
+        .filter(subscription -> Boolean.FALSE.equals(subscription.getEnabled()))
+        .map(
+            subscription -> new SubscriptionStatus().withStatus(SubscriptionStatus.Status.DISABLED))
+        .orElse(null);
+  }
+
+  public List<SubscriptionDestination> listAlertDestinations(UUID subscriptionId) {
+    Optional<EventSubscription> eventSubscriptionOpt =
+        getEventSubscriptionFromScheduledJob(subscriptionId);
+
+    // If the EventSubscription is not found in the scheduled job, retrieve it from the repository
+    EventSubscription eventSubscription =
+        eventSubscriptionOpt.orElseGet(
+            () -> {
+              EntityRepository<? extends EntityInterface> subscriptionRepository =
+                  Entity.getEntityRepository(Entity.EVENT_SUBSCRIPTION);
+
+              return (EventSubscription)
+                  subscriptionRepository.get(
+                      null,
+                      subscriptionId,
+                      subscriptionRepository.getFields("id,destinations,enabled"));
+            });
+
+    if (eventSubscription != null && Boolean.FALSE.equals(eventSubscription.getEnabled())) {
+      return Collections.emptyList();
+    }
+
+    return eventSubscription.getDestinations();
+  }
+
+  public EventSubscriptionDiagnosticInfo getEventSubscriptionDiagnosticInfo(
+      UUID subscriptionId, int limit) {
+    boolean isAllEventsPublished = checkIfPublisherPublishedAllEvents(subscriptionId);
```

Comment:
Any reason why we are checking this here?

Reply:
`isAllEventsPublished` will be part of the `EventSubscriptionDiagnosticInfo`, where we have complete data regarding **pending events and their offsets**. In that data, I thought it would be useful to show a boolean indicating whether all events have been processed. 
EventSubscriptionDiagnosticInfo will have :
1. latest offset
2. current offset
3. isAllEventsPublished
4. unpublishedEventCount
5. unprocessedEvents list

`isAllEventsPublished` can be used by UI if needed.